### PR TITLE
Move guides to the docs folder

### PR DESCRIPTION
**Pull request by fpoli**
This PR moves the guides to the `docs/` folder. The `docs.yml` workflow publishes them.

Let me know if you prefer different folder/file names.

### CODE REVIEW

**Review by Aurel300 — APPROVED** [review]

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: Benchmarks
 on:
   push:
     branches: 'master'
+    paths: '!docs/**'
```
Comment (vakaras) [inline .github/workflows/benchmarks.yml:6]: Now all CI is broken… (it does not start) @fpoli @Aurel300

Comment (fpoli) [reply .github/workflows/benchmarks.yml:6]: I'm on it.

Comment (fpoli) [reply .github/workflows/benchmarks.yml:6]: The correct filter might be `paths-ignore: 'docs/**'`; I'll try.

Comment (fpoli) [reply .github/workflows/benchmarks.yml:6]: @vakaras You can rebase.

Comment (vakaras) [reply .github/workflows/benchmarks.yml:6]: Thanks!

**Pull request merged**